src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1582,7 +1614,14 @@ private void WriteRedRewriter()
                         }
                         else
                         {
-                            WriteLine("      var {0} = ({1})this.Visit(node.{2});", CamelCase(field.Name), field.Type, field.Name);
+                            if (!IsOptional(field) && field.Type != "SyntaxToken")
```

Comment:
`field.Type != "SyntaxToken"` will always be true since the previous if-check checks for "SyntaxToken" already.